src/Msg.elm

```diff
@@ -1,21 +1,54 @@
 module Msg exposing (..)
 
 import Material
-import Model exposing (Model, ExpenseItem)
+import Model exposing (Model, ExpenseForm, ExpenseItem)
+import Result
 
 
 type Msg
     = Noop
-    | AddNewExpense ExpenseItem
+    | AddExpenseForm ExpenseItem
+    | UpdateExpenseAmount String
     | Mdl (Material.Msg Msg)
 
 
+updateExpenseForm : ExpenseForm -> Model -> Model
+updateExpenseForm expenseForm model =
+    let
+        oldUI =
+            model.ui
+
+        newUI =
+            { oldUI | expenseForm = expenseForm }
+    in
+        { model | ui = newUI }
+
+
+updateExpenseFormAmount : String -> Model -> Model
+updateExpenseFormAmount amountInput model =
+    let
+        oldExpenseForm =
+            model.ui.expenseForm
+
+        newExpenseForm =
+            { oldExpenseForm
+                | amountInput = amountInput
+                , amountValue =
+                    String.toFloat amountInput |> Result.withDefault 0.0
+            }
+    in
+        updateExpenseForm newExpenseForm model
```

Comment:
Elm provides a simple way to read data but it lacks on a good way to persist data.